operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -395,22 +399,47 @@ void reconcileIngressControllers() {
         }
     }
 
-    private void createOrEdit(IngressController expected, boolean exists) {
+    private void createOrEdit(IngressController expected, IngressController existing) {
         String name = expected.getMetadata().getName();
 
-        if (exists) {
-            openShiftClient.operator().ingressControllers()
+        if (existing != null) {
+            boolean needsApplied = shouldPatchIngressController(expected, existing);
+            if (needsApplied) {
+                openShiftClient.operator().ingressControllers()
                 .inNamespace(expected.getMetadata().getNamespace())
                 .withName(name)
                 .edit(i -> new IngressControllerBuilder(i)
                         .editMetadata().withLabels(expected.getMetadata().getLabels()).endMetadata()
                         .withSpec(expected.getSpec())
                         .build());
+            }
         } else {
             OperandUtils.createOrUpdate(openShiftClient.operator().ingressControllers(), expected);
         }
     }
 
+    /**
+     * the creation of the expected is dropping the additionalFields (fixed in fabric8 5.10+)
+     * rather than patching all the time, we'll try to detect when the fields we care about have been changed / removed
+     */
+    boolean shouldPatchIngressController(IngressController expected, IngressController existing) {
+        ObjectMapper objectMapper = Serialization.yamlMapper();
+        JsonNode expectedJson = objectMapper.convertValue(expected, JsonNode.class);
+        JsonNode actualJson = objectMapper.convertValue(existing, JsonNode.class);
+        JsonNode patch = JsonDiff.asJson(expectedJson, actualJson);
+        if (patch.isArray()) {
+            int size = patch.size();
+            for (int i = 0; i < size; i++) {
+                JsonNode op = patch.get(i).get("op");
+                if (op != null && !"add".equals(op.asText())) {
```

Comment:
so basically you are expecting no one to change the CR for the ingress controller right? I mean additions, but expect to be modified as removed segments of it

Reply:
We're guarding against any changes here - someone can modify it, but we want to make sure that the spec and label changes we want are there.  If there are additions that's okay - primarily that accounts for the httpEmptyRequestsPolicy field in the spec, which is being dropped from the expected version by https://github.com/fabric8io/kubernetes-client/issues/3570 

If there are modifications or removals though, that means that the current version has changes wrt to our expected version that we want to overwrite.